source/setup/setup.rst

```diff
@@ -2,69 +2,104 @@
 Cerebro Setup
 *************************
 
-Cerebro has a few core dependencies that need to be setup before it can be used.
+We provide a Dockerfile that installs all the dependencies required by Cerebro. Running the Dockerfile will build SCALE-MAMBA and EMP-AGMPC. 
+	* Create the Dockerfile and requirements.txt.
+	        * Dockerfile 
+		::
+
+			FROM rdeng2614/cerebro:initial_image
+			WORKDIR /root
+			RUN git clone https://github.com/mc2-project/cerebro.git
+			WORKDIR /root/cerebro
+			RUN git submodule update --init --recursive
+			WORKDIR /root/cerebro/crypto_backend/emp-toolkit/emp-tool
+			RUN git pull origin master && cmake . && make && sudo make install
+			WORKDIR /root/cerebro/crypto_backend/emp-toolkit/emp-ot
+			RUN cmake . && make && sudo make install
+			WORKDIR /root/cerebro/crypto_backend/emp-toolkit/emp-agmpc
+			RUN git pull origin master
+			RUN cmake . && make
+			WORKDIR /root
+			RUN apt-get update && apt-get install -y \
+				python-gmpy2 \
+				python-pip \
+				emacs
+			COPY requirements.txt /root
+			RUN pip install -r requirements.txt
+
+		* requirements.txt
+		::
+
+			asn1crypto==0.24.0
+			astor==0.8.1
+			astunparse==1.6.3
+			cryptography==2.1.4
+			decorator==4.4.2
+			enum34==1.1.6
+			gmpy2==2.0.8
+			idna==2.6
+			ipaddress==1.0.17
+			keyring==10.6.0
+			keyrings.alt==3.0
+			networkx==2.2
+			numpy==1.16.6
+			ordered-set==3.1.1
+			pathlib==1.0.1
+			pycrypto==2.6.1
+			pygobject==3.26.1
+			pyxdg==0.25
+			SecretStorage==2.3.1
+			six==1.11.0 
+
+	* Place them in the same directory.
+	* In that directory, run ``docker build -t name_of_image``.
+
+After creating the image, run it. You should be in ``/root``, the directory Cerebro was installed in. Complete the setup steps for your desired backend.
 
 SCALE-MAMBA
 ################
 
-The documentation fully detailing SCALE-MAMBA can be obtained by running ``make doc`` in the SCALE-MAMBA repository.
-
-The following is copied from SCALE-MAMBA's current setup documentation.
-
-Prerequisite Libraries:
-*************************
-	* gcc/g++, tested with version 7.2.1
-	* MPIR (compiled with the -cxx flag)
-	* python 2.7.5 (ideally with gmpy2 installed)
-	* OpenSSL (tested with version 1.1.0)
-	* Crypto ++ (tested with version 7.0)
-
-
-
 Setup Steps for SCALE-MAMBA
 *****************************
```

Comment:
For these setup instructions, it would be good to put the instructions in code blocks, instead of in text. This way, a user can just copy and paste the commands into shell and run them.

Reply:
Do you want text/explanations at all? I can remove the text section if you don't want it. I will add a section that has purely shell commands which will set everything up if copy pasted into the shell. Some commands might be a bit cryptic though.